### PR TITLE
211#5 Finalize ASCII colormaps 

### PR DESCRIPTION
**Pull request by mindyls**
Now ready for review - this was fully functional and did return 0.  Updates have been merged into update_products branch, so the final 0 return will be referenced there.

Do not merge until all updates complete for #211.  

# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#211

# Testing Instructions
$GEOIPS/tests/PR_test_update_colormaps.sh

# Summary
* Add plugins/\*\*.txt to pyproject.toml
* Move colormaps from geoips/image_utils/ascii_palettes to
  plugins/txt/ascii_palettes
* Add "configurable" colormap plugin (to allow fully specifying
  via call params)
* Change colormap families to "matplotlib"
* Run pytest on all installed geoips plugin packages

# Output
This returned 0, will additionallyput output in update_products PR once complete, which includes this branch as well.

### CODE REVIEW

File: geoips/image_utils/mpl_utils.py [hunk 1]
```diff
@@ -706,7 +717,10 @@ def create_colorbar(fig, mpl_colors_info):
 
     # Allow arbitrary kwargs to set_label
     set_label_kwargs = {}
-    if "set_label_kwargs" in mpl_colors_info:
+    if (
```
Comment (jsolbrig) [inline geoips/image_utils/mpl_utils.py:720]: Just thinking. I do this too, but I wonder if this kind of thing would be better written as:
```
if mpl_colors_info.get('set_label_kwargs', None):
```

File: geoips/plugins/modules/colormaps/tpw/tpw_pwat.py [hunk 2]
```diff
@@ -95,7 +95,9 @@ def call():
 
     bounds = values + [values[-1] + 1]
     mpl_cmap = from_ascii(
-        pathjoin(gpaths["BASE_PATH"], "image_utils", "ascii_palettes", "tpw_pwat.txt")
+        pathjoin(
+            gpaths["BASE_PATH"], "plugins", "txt", "ascii_palettes", "tpw_pwat.txt"
+        )
```
Comment (jsolbrig) [inline geoips/plugins/modules/colormaps/tpw/tpw_pwat.py:100]: This should probably use `geoips_utils.find_ascii_palette()`?

File: geoips/plugins/modules/colormaps/tpw/tpw_purple.py [hunk 3]
```diff
@@ -48,7 +48,11 @@ def call():
     mpl_colors_info = {
         "cmap": from_ascii(
             pathjoin(
-                gpaths["BASE_PATH"], "image_utils", "ascii_palettes", "tpw_purple.txt"
+                gpaths["BASE_PATH"],
+                "plugins",
+                "txt",
+                "ascii_palettes",
+                "tpw_purple.txt",
```
Comment (jsolbrig) [inline geoips/plugins/modules/colormaps/tpw/tpw_purple.py:55]: This should probably use `geoips_utils.find_ascii_palette()`?

**Review by jsolbrig — APPROVED** [review]

File: geoips/image_utils/mpl_utils.py (hunk 1 above)
Comment (mindyls) [reply geoips/image_utils/mpl_utils.py]: @jsolbrig I updated these merged to update_products - data_fusion passed, and geoips repo tests are currently passing. Please resolve if you are good with this.

File: geoips/plugins/modules/colormaps/tpw/tpw_pwat.py (hunk 2 above)
Comment (mindyls) [reply geoips/plugins/modules/colormaps/tpw/tpw_pwat.py:100]: Going to open Issue for this - do you want to resolve these conversations and we can merge @jsolbrig

File: geoips/plugins/modules/colormaps/tpw/tpw_purple.py (hunk 3 above)
Comment (mindyls) [reply geoips/plugins/modules/colormaps/tpw/tpw_purple.py:55]: https://github.com/NRLMMD-GEOIPS/geoips/issues/226
@jsolbrig Please resolve if you are good with this.

Comment (jsolbrig) [reply geoips/plugins/modules/colormaps/tpw/tpw_purple.py:55]: I think this still should be find_ascii_palette, shouldn't it? As it is, it won't find ascii files from plugin packages.

Comment (jsolbrig) [reply geoips/plugins/modules/colormaps/tpw/tpw_purple.py:55]: As @mindyls pointed out to me in slack, this is a palette that is explicitly in GeoIPS, so this does work.

File: geoips/image_utils/mpl_utils.py (hunk 1 above)
Comment (jsolbrig) [reply geoips/image_utils/mpl_utils.py]: Looks good. Thank you!

**Pull request merged**